### ISSUE TITLE
dist-custom-elements requires a manual load for dependent web components. 

### ISSUE THREAD

**Issue by splitinfinities** [ionic-team/stencil#3033]
**Stencil version:**

```
 @stencil/core@2.7.0
```

**I'm submitting a:**

[ ] bug report
[x] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://stencil-worldwide.herokuapp.com/ or https://forum.ionicframework.com/

**Current behavior:**

Manual code is needed to load specific components that a consumed component is dependent upon when using the dist-custom-elements output target. 

**Expected behavior:**

Components when include should be able to check to see if any dependencies are available, and auto-load them. 

**Steps to reproduce:**

Via @liamdebeasi:
Here is a reproduction: https://github.com/liamdebeasi/ce-build-demo
clone, npm install, npm run serve

This will run the app in production mode which is when treeshaking will happen in vue. Don’t run in dev mode because all of the components will be loaded (no treeshaking)

There’s a single datetime component which uses ion-item, ion-icon, ion-buttons, ion-button, ion-segment, ion-segment-button internally so quite a few ionic components. since it’s CE build, only datetime gets registered and the internal components do not

**Related code:**

via @johnjenkins, shared a naive prototype of this autoloading behavior.  https://github.com/ionic-team/stencil/compare/master...johnjenkins:feat-custom-elements-import-dependencies?expand=1

This may be the solution. In order to confirm, check that code out, and test against Liam’s Reproduction repo: https://github.com/liamdebeasi/ce-build-demo to confirm the related components are loaded and there is no erroneous behavior.

**Other information:**

Current workaround in Ionic is to import the child dependencies manually in your app so that they get registered.

**Comment by kubbafox** [comment]
I am running into the same issue and I would like to get some insight for this approach.

The goal - build all js code into only *one* `my-stencil-script` file for a few different frontend apps - WP (PHP), Angular and Vue. 

The problem - `dist-custom-elements` require dependent web components.

The solution I am testing is to build `outputTargets` as `dist`, then run a different rollup script to bundle everything. However I am not able to get it working end to end. 

My questions are - 
1. If I need to bundle everything, which folder I should check. I am currently using `dist/my-app/my-app.esm.js` file. When I use this file with `inlineDynamicImports: true` in rollup, the output file is not putting back all modules I would expect. 
2. Should I use `loader` folder? I am able to get a larger file and with no console errors if I use `dist/loader/index.js` file, but the application is not loading.

Here is my `rollup.config.js`

```
export default {
  input: 'dist/my-app/my-app.js',
  output: [
    {
      file: 'dist-my-app-es5.js',
      format: 'iife',
      inlineDynamicImports: true,
      sourceMap: 'inline',
    }
  ]
};```

**Comment by rwaskiewicz** [comment]
https://github.com/ionic-team/stencil/pull/3039 resolves this issue

**Comment by arvindanta** [comment]
When will this change be deployed ? . Also , will this be available with https://github.com/ionic-team/stencil-ds-output-targets ?

**Comment by rwaskiewicz** [comment]
I'm going to close this issue, as it has been resolved/launched in v2.9.0 of Stencil.  The team has a ticket in its internal backlog to port this change over to the `dist-custom-elements-bundle` output target
